### PR TITLE
[INLONG-7073][Sort] Support table level metrics for Apache Iceberg connector

### LINKED ISSUE #7073: [Feature][Sort] Support table level metrics for Apache Iceberg connector
### Description

Support table level metrics for Apache Iceberg connector

### Use case

Support table level metrics for Apache Iceberg connector

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by thesumery**
[INLONG-7073][Sort] Support table level metrics for Apache Iceberg connector

### Prepare a Pull Request
- [INLONG-7073][Sort] Support table level metrics for Apache Iceberg connector
- Fixes #7073 

### Motivation

*Support table level metrics for Apache Iceberg connector*

### CODE REVIEW

File: inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/DynamicSchemaHandleOperator.java [hunk 1]
```diff
@@ -182,6 +229,11 @@ private void handleDirtyData(Object dirtyData, JsonNode rootNode, DirtyType dirt
                             .setIdentifier(dirtyOptions.getIdentifier());
                 }
                 dirtySink.invoke(builder.build());
+                if (metricData != null) {
+                    metricData.outputMetricsWithEstimate(
```
Comment (healchow) [inline inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/DynamicSchemaHandleOperator.java:233]: The `outputMetricsWithEstimate` method seems used unsuitable params.

**Review by healchow — APPROVED** [review]

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/sub/SinkTableMetricData.java [hunk 2]
```diff
@@ -228,6 +228,19 @@ public void outputDirtyMetrics(String database, String schema, String table, lon
...
+    /**
+     * output dirty metrics with estimate
+     *
+     * @param database the database name of record
+     * @param schema the schema name of record
+     * @param table the table name of record
+     * @param data the dirty data
+     */
+    public void outputDirtyMetricsWithEstimate(String database, String schema, String table, Object data) {
+        long size = data.toString().getBytes(StandardCharsets.UTF_8).length;
```
Comment (gong) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/sub/SinkTableMetricData.java:240]: I suggest add judging null for data

**Comment by gong** [comment]
lost table level metric restore

**Review by yunqingmoswu — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/sub/SinkTableMetricData.java (hunk 2 above)
Comment (thesumery) [reply inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/metric/sub/SinkTableMetricData.java:240]: ok

**Comment by thesumery** [comment]
> lost table level metric restore

table level metric are in IcebergMultipleStreamWriter line 191，restore table metric at runtime

**Comment by gong** [comment]
> > lost table level metric restore
> 
> table level metric are in IcebergMultipleStreamWriter line 191，restore table metric at runtime

maybe you need call 

> > lost table level metric restore
> 
> table level metric are in IcebergMultipleStreamWriter line 191，restore table metric at runtime

maybe you need call SinkTableMetricData#registerSubMetricsGroup for table level metric restore

**Comment by thesumery** [comment]
> registerSubMetricsGroup

It‘s different from other multiple sink connector.It's a 'IcebergMultipleStreamWriter' include many  'IcebergSingleStreamWriter'，so SinkTableMetricData#registerSubMetricsGroup it not suitable for this.

**Review by gong — APPROVED** [review]

**Pull request merged**